dynamiqs/sesolve/propagator.py

```diff
@@ -1,10 +1,17 @@
+from functools import lru_cache
+
 import torch
 from torch import Tensor
 
 from ..solvers.propagator import Propagator
 
 
 class SEPropagator(Propagator):
+    @lru_cache(maxsize=1)
+    def propagator(self, t: float, delta_t: float) -> Tensor:
```

Comment:
Actually, this suffers from the same issue as #129, it will be recomputed for every new `t`.